lib/p2p/packets/Packet.ts

```diff
@@ -56,15 +58,20 @@ abstract class Packet<T = any> implements PacketInterface {
   constructor(bodyOrPacket?: T | PacketInterface, reqId?: string) {
     if (isPacketInterface(bodyOrPacket)) {
       // we are deserializing a received packet from a raw JSON string
-      this.body = bodyOrPacket.body;
+      delete bodyOrPacket.header.type; // deleting type from header because it's being used only for the wire protocol
       this.header = bodyOrPacket.header;
+
+      if (bodyOrPacket.body) {
+        this.body = bodyOrPacket.body;
```

Comment:
What's the advantage of this over `this.body = bodyOrPacket.body`? I think it winds up undefined either way when the body is undefined.

Reply:
if it's undefined, then the property is set to undefined. it's not the same as if the property wouldn't be set at all. So if doing a deep equal check on the objects, it fails. It's possible to just remove the undefined properties before doing the equal check, but I thought that this would be better.

Reply:
Ah yeah that's what you meant by deep checks, ok cool, thanks.